File: dart/test/grammar_test.dart
import 'package:test/test.dart';
import 'package:squirrel_parser/squirrel_parser.dart';

void main() {
  test('parses simple grammar', () {
    const String simpleGrammar = '''
      Test <- "hello";
    ''';

    expect(
      () => MetaGrammar.parseGrammar(simpleGrammar),
      returnsNormally,
    );
  });

  test('parses multiline grammar', () {
    const String multilineGrammar = '''
      JSON <- Value;
      Value <- "test";
    ''';

    expect(
      () => MetaGrammar.parseGrammar(multilineGrammar),
      returnsNormally,
    );
  });
}

File: typescript/tests/grammar.test.ts
import { MetaGrammar } from '../src/index.js';

describe('Grammar parsing', () => {
  it('parses simple grammar', () => {
    const simpleGrammar = `
      Test <- "hello";
    `;

    expect(() => MetaGrammar.parseGrammar(simpleGrammar)).not.toThrow();
  });

  it('parses multiline grammar', () => {
    const multilineGrammar = `
      JSON <- Value;
      Value <- "test";
    `;

    expect(() => MetaGrammar.parseGrammar(multilineGrammar)).not.toThrow();
  });
});
